packages/common-all/src/engine/EngineV3Base.ts

```diff
@@ -43,7 +46,8 @@ export abstract class EngineV3Base implements ReducedDEngine {
 
   constructor(
     protected noteStore: INoteStore<string>,
-    protected logger: DLogger
+    protected logger: DLogger,
+    public vaults: DVault[]
```

Comment:
I'm thinking it might be better to make vaults `protected` instead of `public`.  Not sure if using the engine is the right place to get `vaults`.  Is this possible, or does it lead to a snowball effect of changes?

Reply:
Right now its part of `DCommonProps` so changing to `protected` isn't possible w/o another refactoring :/